src/main/java/seedu/address/model/stats/HistogramStat.java

```diff
@@ -0,0 +1,139 @@
+     */
+    public HistogramStat(ObservableList<Buyer> buyerList, ObservableList<Property> propertyList,
+                         boolean showBuyer, boolean showProperty, String titleArgs) {
+        this.buyerList = buyerList;
+        this.propertyList = propertyList;
+        this.showBuyer = showBuyer;
+        this.showProperty = showProperty;
+        this.titleArgs = titleArgs;
+    }
+
+    /**
+     * Returns a two element array containing the min in index 0, and max in index 1.
+     */
+    private double[] getGlobalMinMax() {
+        double buyerMin = buyerList.stream()
+                .min((x, y) -> x.getPrice().compareTo(y.getPrice())).map(p -> p.getPrice().value).get();
+        double buyerMax = buyerList.stream()
+                .max((x, y) -> x.getPrice().compareTo(y.getPrice())).map(p -> p.getPrice().value).get();
+        double propertyMin = propertyList.stream()
+                .min((x, y) -> x.getPrice().compareTo(y.getPrice())).map(p -> p.getPrice().value).get();
+        double propertyMax = propertyList.stream()
+                .max((x, y) -> x.getPrice().compareTo(y.getPrice())).map(p -> p.getPrice().value).get();
+        double[] minmax = new double[2];
+        minmax[0] = (int) (Math.min(buyerMin, propertyMin) / 100000) * 100000;
+        minmax[1] = (int) (Math.max(buyerMax, propertyMax) / 100000 + 1) * 100000;
+        return minmax;
+    }
+
+    /**
+     * Returns the correct bin for this price.
+     */
+    private int getBin(double price, double min, double interval) {
+        return (int) Math.min((price - min) / interval, BIN_COUNT - 1);
+    }
+
+    /**
+     * Returns the label for the bin of this interval.
+     */
+    private String getRangeLabel(double price) {
+        if (price >= MILLION) {
+            price /= MILLION;
+            return String.format("%.2fM", price);
+        } else {
+            price /= THOUSAND;
+            return String.format("%dK", (long) price);
+        }
+    }
+
+    /**
+     *  Adds this list to the dataset.
+     */
+    private void addToDataSet(DefaultCategoryDataset dataset,
+                              ObservableList<? extends Pricable> list, String key,
+                              double min, double max) {
+        double interval = (max - min) / BIN_COUNT;
+        double binVal = min;
+        int[] bins = new int[BIN_COUNT];
+        list.stream()
+                .forEach(x -> bins[getBin(x.getPrice().value, min, interval)]++);
+        for (int i = 0; i < bins.length; i++, binVal += interval) {
+            String label = String.format("%s -- %s", getRangeLabel(binVal), getRangeLabel(binVal + interval));
+            dataset.addValue(bins[i], key, label);
+        }
+    }
+
+    /**
+     * Creates a histogram to displayed as a popup.
+     * Shows either buyers, properties or both.
+     *
+     * @return price histogram.
+     */
+    public JFreeChart create() {
+        assert !buyerList.isEmpty() && !propertyList.isEmpty();
```

Comment:
btw this assertion fails when we try to do `stat buyer` with no `properties`
or `stat` with no `properties`